README.md

```diff
@@ -3,21 +3,108 @@
+It handles HTTP and WebSocket connections towards the device and forwards back the reply.
 
-  * Run `mix setup` to install and setup dependencies
-  * Start Phoenix endpoint with `mix phx.server` or inside IEx with `iex -S mix phx.server`
+It has currently only been tested with [ttyd](https://github.com/tsl0922/ttyd),
+but _should_ work for arbitrary requests,
+as long as you keep in mind the [limitations](#known-issues-and-limitations).
 
-Now you can visit [`localhost:4000`](http://localhost:4000) from your browser.
+The connection is based on a token system, where the device and the user both connect to the server
+with the same connection token.
 
-Ready to run in production? Please [check our deployment guides](https://hexdocs.pm/phoenix/deployment.html).
+## Running the Forwarder locally
 
-## Learn more
+### Start the server
 
-  * Official website: https://www.phoenixframework.org/
-  * Guides: https://hexdocs.pm/phoenix/overview.html
-  * Docs: https://hexdocs.pm/phoenix
-  * Forum: https://elixirforum.com/c/phoenix-forum
-  * Source: https://github.com/phoenixframework/phoenix
+The first component to be available should be this server,
+as both the device and the user will connect to it.
+
+#### Manual installation
+
+Clone this repository locally and navigate to it using
+
+```bash
+git clone https://github.com/edgehog-device-manager/edgehog_device_forwarder && \
+cd edgehog_device_forwarder
+```
+
+> At the time of writing, not all the relevant code has been merged to the main branch.
+> Use this command instead
+> ```bash
+> git clone https://github.com/noaccOS/edgehog_device_forwarder \
+> -b feat/fallback-controller && cd edgehog_device_forwarder
+> ```
+
+You will need to install Elixir 1.15 with OTP 26.
+
+> If you use [`asdf`](https://asdf-vm.com), you can use it to install our officially supported
+> version, by using the `.tool-versions` file from
+> [edgehog's repository](https://github.com/edgehog-device-manager/edgehog/blob/main/.tool-versions)
+> and copy it here.
+
+Now use this command to start the server:
+
+```bash
+mix phx.server
+```
+
+### Connect a device
+
+You will now need to start the device process.
+You can do so from the same machine or another device or virtual machine, the process should look
+exactly the same (granted you open up port 4000 on the server's firewall).
+
+To connect to the server we'll use a special program in
+[`edgehog-device-runtime`](https://github.com/edgehog-device-manager/edgehog-device-runtime),
+`e2e-test-forwarder`.
+
+#### Manual install
+
+> At the time of writing, you need to use a fork of said repository, because not all the code
+> has been merged.
+> Use the command
+>
+> ```bash
+> git clone https://github.com/rgallor/edgehog-device-runtime -b ws-over-ws && \
+> cd edgehog-device-runtime/e2e-test-forwarder
+> ```
+
+This is a rust program, so to compile it you will need to install `cargo`.
+
+```bash
+cargo run -- -H server.local -t abcd
+```
+
+In this example
```

Comment:
...to install cargo. Then you can run it by using the following command:
```bash
cargo run -- -H <host> -t <token>
```
For instance:
```bash
cargo run -- -H server.local -t abcd
```
where: